src/app/model/User.ts

```diff
@@ -35,6 +38,20 @@ const defaultUserData: UserInterface = {
 class User extends BaseModel {
   VolunteerStatus = VolunteerStatus;
 
+  async saveNewUser(data: UserInterface) {
+    const id = uuidV4(); //generate User id  TODO: Setup firebase tables to autogenerate ids
```

Comment:
If i recall, user IDs generated through firebase were just auto incrementing numbers. @qdozaq, Do you know if that's true?

Reply:
![image](https://user-images.githubusercontent.com/3273671/79700294-f6773680-8294-11ea-8bb7-8ad2ea718f25.png)

Reply:
Ok, about user ID, once a user sign up through auth, they got an id, this is stored inside firebase.auth.uid

This uid should be used as the key 

This can be obtain by during sigun up step, you can get the result.user -> set this inside the users collection
![image](https://user-images.githubusercontent.com/3273671/79700358-5372ec80-8295-11ea-8d2e-41c0fe9e659f.png)
Link for ref:  https://firebase.google.com/docs/auth/web/phone-auth

Reply:
@mat10tng 
Ok, it is easy to get the uid from the phone auth which I can use later when saving the profile.  

But I am not sure about the next step after PhoneAuth in the signup process -- **Connect social media**.  As of now we are 'authenticating' again and then we get back an entirely different uid from the one we get from phone auth.

I think I will have to research more about [linking multiple auths ](https://firebase.google.com/docs/auth/web/account-linking)-- or depening on the rush of this pr maybe we just ignore it for now?  Especially since @qdozaq said we're doing a complete overhaul of firebase...

One more thing, I noticed the format of the uuid provided from the auth:
`IpENTHiC3bfPB6h6CNsWZ2izC123
`
 look different from the uuid generated by that `uuidv4():
`1e8d498f-27b3-4972-89e3-ee07a31ea183
`

Reply:
no worry, the auth.uid is always the same.

Reply:
I'm getting 2 different uuid's -- one for the phone auth and one for the social media auth.
from phone auth response:
![Screen Shot 2020-04-19 at 4 34 12 PM](https://user-images.githubusercontent.com/19916961/79702756-fc045a80-825b-11ea-87f7-d715625b18a4.png)

from google auth response:
![Screen Shot 2020-04-19 at 4 34 56 PM](https://user-images.githubusercontent.com/19916961/79702763-07578600-825c-11ea-9986-cf01ee581023.png)

This is using the auth functions we had in place before I started this.  I think we need to somehow pass the uid we get from the phone auth to the social media auth call, but I would need to look more into that.

Reply:
After conversation with @mat10tng we've decided to not focus on solving this issue right now so we can work on other things for mvp0.

What we have now will authenticate phone and connect that with the user profile information filled out by the new user.  It will also authenticate the social media if chosen and retain the email/displayName from that. 

We've created [this ticket](https://github.com/factn/resilience-app/issues/290) to track open items / edge cases around the Signup flow.